model-datastructure/src/commonMain/kotlin/org/modelix/model/lazy/NonCachingObjectStore.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2023.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.lazy
+
+import org.modelix.model.IKeyValueStore
+
+class NonCachingObjectStore(override val keyValueStore: IKeyValueStore) : IDeserializingKeyValueStore {
+
+    override fun <T> getAll(hashes_: Iterable<String>, deserializer: (String, String) -> T): Iterable<T> {
+        val hashes = hashes_.toList()
```

Comment:
Can we iterate over the result map?

```kt
    override fun <T> getAll(hashes: Iterable<String>, deserializer: (String, String) -> T): Iterable<T> {
        val serialized: Map<String, String?> = keyValueStore.getAll(hashes)
        return serialized.mapNotNull { (hash, value) -> value?.let { deserializer(hash, it) } }
```

The cast to `as T` would be bad, if  any requested `hash` is not in `serialized.values` or if any of `serialized.keys` is `null`.

Reply:
The result is expected to be in the same order as the `hashes`. That `IDeserializingKeyValueStore` API is poorly designed and may be removed completely in the future.

I added a not null check.